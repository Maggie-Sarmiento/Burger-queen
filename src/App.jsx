import React, { useState } from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import LogIn from './No-Authenticate/Login/LogIn';
import Staff from './Authenticate/Staff';
import { auth, db } from './Firebase/firebase.config';
import { onAuthStateChanged } from 'firebase/auth';
import { getDoc,doc } from 'firebase/firestore';
import './App.css'; 
import NotFound from './No-Authenticate/NotFound';

const App = () => {

  const [authenticate, setAuthenticate] = useState(null);
  const [rol, setRol] = useState(null);

  onAuthStateChanged(auth, (user) => {
    if (user){
      console.log(authenticate)
      setAuthenticate(user);
      updateRol(user);
    } else {
      setAuthenticate(null);
    }
  });

  const getRol = async(uid) => {
    const ref = doc(db, `roles/${uid}`);
    const docSnap = await getDoc(ref);
    return docSnap.data().rol;
  }

  const updateRol = (user) => {
    getRol(user.uid)
    .then((rol) => {
      setRol(rol);
    }) .catch((error)=>{
      console.log(error);
    })
  }
  console.log(rol);
  return (
    <Router> {
      authenticate
      ?
      <Routes>
        <Route path='/' element={<LogIn />} />
        <Route path='/staff' element={<Staff rol={rol} />} />
      </Routes>
      :
      <Routes>
        <Route path='/' element={<LogIn />} />
        <Route path='*' element={<NotFound />} />
      </Routes>
      }
    </Router>
  );
}

export default App;
