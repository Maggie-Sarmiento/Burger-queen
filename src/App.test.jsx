/* eslint-disable no-undef */
// import { render, screen } from '@testing-library/react';
import React from 'react';
import { create, act } from 'react-test-renderer';
import App from './App';

describe.skip('test App', () => {
  test('render App', async () => {
    let root;
    act(() => {
      root = create(
        <App />,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
