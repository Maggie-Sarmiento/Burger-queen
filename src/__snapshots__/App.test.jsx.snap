// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App render App 1`] = `
<section
  className="login"
>
  <img
    alt="imgLogin"
    className="imgLogin"
    src="https://svgshare.com/i/gjb.svg"
  />
  <section
    className="sectionLogin"
  >
    <div
      className="welcome"
    >
      <img
        alt="logoBQ"
        className="imgLog"
        src="https://svgshare.com/i/gmq.svg"
      />
      <p
        className="slogan"
      >
        Inicia sesión para administar el staff y menús o para tomar la orden de los clientes.
      </p>
    </div>
    <form
      className="formLogin"
    >
      <input
        className="inputEmail"
        onChange={[Function]}
        placeholder="Ejemplo@gmail.com"
        type="email"
      />
      <input
        className="inputPass"
        onChange={[Function]}
        placeholder="Contraseña"
        type="password"
      />
      <button
        className="btnLogin"
        onClick={[Function]}
        type="button"
      >
        Iniciar sesión
      </button>
      <span
        className="alertError"
      />
    </form>
  </section>
</section>
`;
